rust/lance/src/dataset/scanner.rs

```diff
@@ -831,6 +835,10 @@ impl Scanner {
             // This tests if any of the fragments are missing the physical_rows property (old style)
             // If they are then we cannot use scalar indices
             if filter_plan.index_query.is_some() {
+                // Index scan doesn't honor the fragment allowlist today.
+                // TODO: we could filter the index scan results to only include the allowed fragments.
+                self.ensure_not_fragment_scan()?;
+
```

Comment:
Can you move this to this spot (it's line 1539 on main right now I think):

```
            (Some(index_query), None, true) => {
                // The filter is completely satisfied by the index.  We
                // only need to search the index to determine the valid row
                // ids.
                let index_query = Arc::new(ScalarIndexExec::new(
                    self.dataset.clone(),
                    index_query.clone(),
                ));
                PreFilterSource::ScalarIndexQuery(index_query)
            }
```

The only node that doesn't handle a fragment scan is the `ScalarIndexQuery` node.  So right here is a bit too restrictive.  If we do that then we're good to go.

Reply:
Moved to line 1539. PTAL.